internal/host/docker_unix.go

```diff
@@ -55,9 +52,13 @@ func (d Docker) Run(ctx context.Context, cmd host.Cmd) (host.WaitStatus, error)
 	if cmd.Stdin != nil {
 		args = append(args, "--interactive")
 	}
-	args = append(args, []string{"--user", d.User}...)
+
+	dockerConnectionUser := strings.Split(d.Connection, "@")[0]
+	dockerConnectionContainer := strings.Split(d.Connection, "@")[1]
```

Comment:
This can panic: if a user gives invalid input (eg: `container_name_only`), the length of `strings.Split(d.Connection, "@")` will be 1, and this will panic.

Originally, we had user/group default to `0:0` (root). I think we could keep this, as most cases, we'd be running as root. So, the format would be: `[<name|uid>[:<group|gid>@]<image>`.

I'm also thinking whether it is best to move the logic to parse this separately, whether than here on `NewDocker`...